packages/wallets/alpha/src/adapter.ts

```diff
@@ -201,13 +209,20 @@ export class AlphaWalletAdapter extends BaseMessageSignerWalletAdapter {
         }
     }
 
-    async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
+    async signAllTransactions<T extends Transaction | VersionedTransaction>(transactions: T[]): Promise<T[]> {
         try {
             const wallet = this._wallet;
             if (!wallet) throw new WalletNotConnectedError();
 
+            for (const transaction of transactions) {
+                if (isVersionedTransaction(transaction))
+                    throw new WalletSendTransactionError(
+                        `Signing versioned transactions isn't supported by this wallet`
+                    );
+            }
+
             try {
-                return (await wallet.signAllTransactions(transactions)) || transactions;
+                return ((await wallet.signAllTransactions(transactions as Transaction[])) as T[]) || transactions;
```

Comment:
All the adapters are broken now except this one, which I refactored to check out practicability of the design. This function (and to a lesser degree, `signTransaction`) isn't great. This can be better enforced with types rather than runtime checks, and there's too much in the way of type guards and casting required currently.